src/components/Editor/components/ProfilePanel.js

```diff
@@ -9,17 +11,111 @@ import firebase from "firebase";
  * size: number? representing the pixel width of the panel
  */
 
+const filter = new Filter();
+
 class ProfilePanel extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       prevWidth: 0,
       width: this.props.width,
+      isHovering: false,
+      editing: false,
+      name: this.props.displayName,
+      displayNameMessage: "",
     };
   }
 
   componentDidUpdate() {}
 
+  handleEditClick = () => {
+    this.setState(prevState => {
+      return { editing: true };
+    });
+  };
+
+  onChange = e => {
+    this.setState({ name: e.target.value });
+  };
+
+  checkInputs = () => {
+    const name = this.state.name;
+    if (name.length < MINIMUM_DISPLAY_NAME_LENGTH || name.length > MAXIMUM_DISPLAY_NAME_LENGTH) {
+      this.setState({
+        displayNameMessage: `Display name must be between ${MINIMUM_DISPLAY_NAME_LENGTH}-${MAXIMUM_DISPLAY_NAME_LENGTH} characters long`,
+      });
+      return true;
+    } else if (name.match(/[^a-zA-Z0-9!@#$%]/)) {
```

Comment:
Add a space inbetween the brackets like [^a-zA-Z0-9!@#$% ] to make it so that spaces are allowed in the display name and change the error message below to reflect that